src/markdown/odoc_md.ml

```diff
@@ -0,0 +1,86 @@
+(* This exe will compile a markdown file, outputting a compiled `page-x.odoc` file.
+   This is tightly coupled with the internal representation of odoc files and thus needs
+   to be run with the exact same version of odoc that it is compiled with. *)
+
+open Odoc_model
+
+let parse id input_s =
+  let location =
+    Lexing.{ pos_fname = input_s; pos_lnum = 1; pos_cnum = 0; pos_bol = 0 }
+  in
+  let str = In_channel.(with_open_bin input_s input_all) in
+  let content, _warnings = Doc_of_md.parse_comment ~location ~text:str () in
+  let content, () =
+    Semantics.ast_to_comment ~internal_tags:Expect_none ~sections_allowed:`All
+      ~tags_allowed:true
+      ~parent_of_sections:(id :> Paths.Identifier.LabelParent.t)
+      content []
+    |> Error.raise_warnings
```

Comment:
Are we not interested in these warnings at all? we could do:
```diff
-  let content, _warnings = Doc_of_md.parse_comment ~location ~text:str () in
-  let content, () =
+  let content, parser_warnings =
+    Doc_of_md.parse_comment ~location ~text:str ()
+  in
+  let (content, ()), semantics_warnings =
     Semantics.ast_to_comment ~internal_tags:Expect_none ~sections_allowed:`All
       ~tags_allowed:true
       ~parent_of_sections:(id :> Paths.Identifier.LabelParent.t)
       content []
-    |> Error.raise_warnings
+    |> Error.unpack_warnings
   in
-  content
+  (content, List.map Error.t_of_parser_t parser_warnings @ semantics_warnings)
```
and pass them to `Odoc_odoc.Odoc_file.save_page`

Reply:
Sure, why not.